test/txRelay.js

```diff
@@ -326,14 +326,101 @@ contract('TxRelay', (accounts) => {
+
+      assert.isTrue(await testTxRelay.whitelist.call(1, sender2), "sender2 should be on list")
+      assert.isTrue(await testTxRelay.whitelist.call(1, sender3), "sender3 should be on list")
+
+      assert.isFalse(await txRelay.whitelist.call(1, sender1), "sender1 should not be on list")
+    })
+
+    it("Should allow owner of list to add another sender", async () => {
+      await testTxRelay.addToWhitelist(1, [sender4], {from: sender1})
+
+      assert.isTrue(await testTxRelay.whitelist.call(1, sender2), "sender2 should be on list")
+      assert.isTrue(await testTxRelay.whitelist.call(1, sender3), "sender3 should be on list")
+      assert.isTrue(await testTxRelay.whitelist.call(1, sender4), "sender4 should be on list")
+    })
+
+    it("Should not allow non-owner to add another sender", async () => {
+      try {
+        await testTxRelay.addToWhitelist(1, [sender5], {from: sender2})
+      } catch (e) {
+        assert.match(e.message, /revert/, "should have thrown")
+        errorThrown = true
+      }
+      assertThrown(errorThrown, "Should have thrown")
+    })
+
+    it("Should allow owner of list to remove senders", async () => {
+      await testTxRelay.removeFromWhitelist(1, [sender4, sender3], {from: sender1})
+
+      assert.isTrue(await testTxRelay.whitelist.call(1, sender2), "sender2 should be on list")
+      assert.isFalse(await testTxRelay.whitelist.call(1, sender3), "sender3 not should be on list")
+      assert.isFalse(await testTxRelay.whitelist.call(1, sender4), "sender4 not should be on list")
+    })
+
+    it("Removing sender not on list should not have an effect", async () => {
+      assert.isFalse(await testTxRelay.whitelist.call(1, sender3), "sender3 not should be on list")
+
+      await testTxRelay.removeFromWhitelist(1, [sender3], {from: sender1})
+
+      assert.isFalse(await testTxRelay.whitelist.call(1, sender3), "sender3 not should be on list")
+    })
+
+    it("Should not allow non-owner to remove sender", async () => {
+      try {
+        await testTxRelay.removeFromWhitelist(1, [sender2], {from: sender2})
+      } catch (e) {
+        assert.match(e.message, /revert/, "should have thrown")
+        errorThrown = true
+      }
+      assertThrown(errorThrown, "Should have thrown")
+    })
+
+    it("Should allow non-owner to create their own whitelist", async () => {
+      await testTxRelay.addToWhitelist(2, [sender5, sender6], {from: sender2})
+
+      assert.isFalse(await testTxRelay.whitelist.call(2, sender2), "sender2 should not be on list")
+      assert.isFalse(await testTxRelay.whitelist.call(2, sender3), "sender3 should not be on list")
+      assert.isFalse(await testTxRelay.whitelist.call(2, sender4), "sender4 should not be on list")
+      assert.isTrue(await testTxRelay.whitelist.call(2, sender5), "sender5 should be on list")
+      assert.isTrue(await testTxRelay.whitelist.call(2, sender6), "sender6 should be on list")
```

Comment:
Can you add tests making sure that if the user specifies a white-list other than 0, this the sender must be on the whitelist? Both with a sender who is on the whitelist and someone who is not on the whitelist (so it succeeds and fails respectively).

This requires the changes to `signPayload` requested above.

Reply:
Oh, damn. Totally forgot about this!

Reply:
Should be addressed now